### ISSUE TITLE
jsHint Support in Gulp

### ISSUE THREAD

**Issue by desaiuditd** [digisavvy/some-like-it-neat#26]
I see there are tasks defined for jsHint in `Gulpfile.js` [here](https://github.com/digisavvy/some-like-it-neat/blob/master/gulpfile.js#L84-L85). but when I un-comment them, it tries to evaluate the JS files from vendor i.e., third party JS files that are included in the theme. We do not want to find errors in others' code. At least I would prefer that.

We could separate out jsHint tasks from the concatenation task. So that we just pass our JS files to jsHint for evaluation. Also with that, a common directory can be defined let's say `app` or `frontend` which will hold theme's JS files. So we can pass that particular directory only to jsHint. So other JS files from third party libraries won't come into picture.

I've this fixed here eb60d1738cbf93a5640cb7626eb2e8f2cd3f56df. I could send a PR if it is appropriate to merge and useful to others.

Note: We are also thinking to integrate PHP Code Sniffer with the theme and other Dev tools. I'll be working on those tasks in the forked repo. We are also following SMACSS in development which suits the theme.

Any thoughts / suggestions ?

**Comment by digisavvy** [comment]
Hi there,
Thanks so much for the thoughtful post and I apologize for the delay. So, I'm totally in agreement with you, regarding JShint. That part had stumped me a bit, in truth.  I would totally welcome a PR for this change and also for your other proposed changes, too. I would ask you submit them on the development branch, if you wouldn't mind: https://github.com/digisavvy/some-like-it-neat/tree/development/ 

Again, many thanks.

**Comment by desaiuditd** [comment]
Wow !

That's great. I'll be upto it and prepare a PR. Will submit in a couple of days.

Just putting one of my colleague in the loop. @jeffreynolte

Thanks.

**Comment by jeffreynolte** [comment]
Awesome news @desaiuditd @digisavvy!  @digisavvy - Great work with the theme, we also follow a similar workflow at Moxie and are planning to implement a bunch of features outlined in the roadmap for a [Yeoman generator](https://github.com/moxienyc/generator-moxie-wp#roadmap).  Would be great to continue collaboration.

**Comment by digisavvy** [comment]
Super stoked to hear this! I'd been eyeballing Yeoman as well. Was planning on getting dirty with it this weekend, if time allows. I've been trying to hammer down accessibility and i18n stuff and I'm getting close.

**Comment by digisavvy** [comment]
Hey folks, 
Thanks again for the PR! =)

I had some issues merging it into my repo for some reason (let's just call it user error on my part), so I manually grabbed the changes from jshint you sent and placed into the dev branch. I tested and it looks stable: https://github.com/digisavvy/some-like-it-neat/blob/development/gulpfile.js - There were some other unrelated changes I didn't take such as color attributes and such, I was just grabbing the most important one. =)

So if you wanted to take a look, the theme's been updated: https://github.com/digisavvy/some-like-it-neat/blob/development/gulpfile.js
